### ISSUE TITLE
IAM Role ${App}-executionrole is missing an allowed ecr:UntagResource action

### ISSUE THREAD

**Issue by camilosantana** [aws/copilot-cli#2247]
When I try to delete a service via `copilot svc delete`, the first attempt fails. Running it a second time succeeds.

Checking events in the infrastructure stack, I see the following error:

`logical ID ECRRepo<all service repos; one line per>`

```shell
Resource handler returned message: "User: arn:aws:sts::##:assumed-role/app-executionrole/...
 is not authorized to perform: ecr:UntagResource 
 on resource: arn:aws:ecr:us-east-1:##:repository/app/service (Service: Ecr, Status Code: 400,
 Request ID: ...., HandlerErrorCode: GeneralServiceException)
--
```

checking the Inline policy on the above role, I see:

```json
"Action": [
    "ecr:DescribeImageScanFindings",
    "ecr:GetLifecyclePolicyPreview",
    "ecr:CreateRepository",
    "ecr:GetDownloadUrlForLayer",
    "ecr:GetAuthorizationToken",
    "ecr:ListTagsForResource",
    "ecr:ListImages",
    "ecr:DeleteLifecyclePolicy",
    "ecr:DeleteRepository",
    "ecr:SetRepositoryPolicy",
    "ecr:BatchGetImage",
    "ecr:DescribeImages",
    "ecr:DescribeRepositories",
    "ecr:BatchCheckLayerAvailability",
    "ecr:GetRepositoryPolicy",
    "ecr:GetLifecyclePolicy",
    "ecr:TagResource"
],
```

and note that `ecr:UntagResource` is not in the list.

**Comment by camilosantana** [comment]
would updating [this template](https://github.com/aws/copilot-cli/blob/23bbb9579213e4820db19049fe9b20fb054c07da/templates/app/versions/v1.0.1/app.yml#L86) resolve the issue?

**Comment by camilosantana** [comment]
i manually added `ecr:UntagResource` to the execution role's inline policy. i'm able to delete services without issue.

```
?1 oula % copilot svc delete                                                                                                                  
Service name:worker
Are you sure you want to deleteworker from application portal? Yes
✔ Deleted service worker from environment staging.
✔ Deleted resources of service worker from application portal.
✔ Deleted service worker from application portal.
Recommended follow-up actions:
- Run `copilot pipeline update` to update the corresponding pipeline if it exists.
```

**Comment by rverpillot** [comment]
Same error here.

Conditions to reproduce this issue:
- deploying more than one service.
- Using custom tags (`copilot app init --resource-tags`).

When the infrastructure CF stack is updated to remove ECR repository, the custom tags are missed (bug) and all AWS resources are updated to remove these tags.

**Comment by iamhopaul123** [comment]
Hello @rverpillot. Unfortunately I can't reproduce the issue 🥲

> When the infrastructure CF stack is updated to remove ECR repository, the custom tags are missed (bug) and all AWS resources are updated to remove these tags.

Could you please elaborate on this? And the custom tags were there before the stack deletion fails? How about the other Copilot auto created tags?

**Comment by rverpillot** [comment]
Hello @iamhopaul123.

Custom tags are located in the tags part of the infrastructure CF stack. 
![image](https://user-images.githubusercontent.com/34237486/168752268-d7905b7b-a43c-4934-9d0d-33cbf1cb4067.png)

When deleting a service, this stack is updated to remove its ECR repository. But all custom tags are missing, only the copilot tag `copilot-application` is remaining. Because of this modification, the remaining AWS resources of the stack are updated to remove these tags, but removal fails on ECR because policy `ecr:UntagResource` is missing.
![image](https://user-images.githubusercontent.com/34237486/168751953-36e568fe-1445-42c0-9a94-e8de834c9aff.png)

Hope this can help.

**Comment by iamhopaul123** [comment]
Hello @rverpillot. Can you help me to check what the IAM role for the failed stack is?
<img width="864" alt="Screen Shot 2022-05-17 at 12 47 12 PM" src="https://user-images.githubusercontent.com/51377557/168897592-14ba85e1-6314-4aa6-93fe-268106fb3fb6.png">

**Comment by rverpillot** [comment]
@iamhopaul123 I found the bug: the field `AdditionalTags` is missing in the struct `appConfig`. 
Add it fixes this issue.
https://github.com/aws/copilot-cli/blob/22f6b76d0f1a11c79a2edb3b15b47ac32b967fd2/internal/pkg/deploy/cloudformation/app.go#L302-L307
```
	appConfig := stack.NewAppStackConfig(&deploy.CreateAppInput{
		Name:           app.Name,
		AccountID:      app.AccountID,
		AdditionalTags: app.Tags,
		Version:        deploy.LatestAppTemplateVersion,
	})
```
I don't understand why you can't reproduce this issue. Are you sure you deployed 2 or more services ?

**Comment by iamhopaul123** [comment]
I see. So the error didn't happen when removing the ECR repo but updating when one of the service was removed from the app. That totally makes sense to me. Thank you for finding the issue! I'll send out a quick fix soon.

**Comment by huanjani** [comment]
This fix is now released in v1.19.0: https://github.com/aws/copilot-cli/releases/tag/v1.19.0! 🎉
